app/core/RPCMethods/wallet_switchEthereumChain.js

```
@@ -85,10 +85,16 @@ const wallet_switchEthereumChain = async ({
     return;
   }
 
-  throw providerErrors.custom({
```

It seems like there has been a misunderstanding about what this was intended to do. This error is for the _dapp_, not us. It's important that we tell the dapp that the request failed, and _why_ it failed.

---

100% agree with @Gudahtt, this is the big difficulty here.
We HAVE to throw an error on the provider for the dapp to get the error feedback.

>[!warning]
> Removing this error throw is going to break all the dapps workflows !

The side effect is, as we have a Sentry error catch-all at the top of the app, every thrown error that is not caught in try/catch is also considered as an unhandled error by Sentry catch-all. And it's reported to sentry.

Here, to me, we should have the error reported to the dapp only, no Sentry, and optionally if we really want to track them, have a Segment tracking event.

But given that it's totally out of our scope and we can't do anything about it, I'm not sure we should track that. It would only tell us that dapps are generating errors. Maybe we could monitor what dapp throws the more errors and contact them, but is this really our problem?

---

@frankvonhoven is there any way to throw the error for the dapp without throwing it to Sentry? If yes we should go for that option.

---

That's the question, afaik when we throw an Error, it is caught by the error boundary, and reported to Sentry, so whenever we throw an error in the app, it will always get reported to sentry and the user will see the Error Screen. But we could check for that particular error, and perhaps prevent it from going to Sentry. But then we're going to be responsible for maintaining a block list. Perhaps we could silence that error in Sentry? or is it a frequent occurrence that it is polluting Sentry?

---

It is noisy in Sentry: 820k over the past 30 days.
Do we know what are the pros/cons of maintaining a blocklist in the client VS silencing the error in Sentry?

---

We could add this error to the [custom error handler](https://github.com/MetaMask/metamask-mobile/blob/d01cf65b91aba194db789df0224cd029cf554ff8/app/core/ErrorHandler/ErrorHandler.ts)

To improve the maintainability of this handler, we could adopt an error message pattern that matches the discard mechanism filter.